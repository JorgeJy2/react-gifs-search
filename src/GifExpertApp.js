
import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GiftGrid } from './components/GiftGrid';

export const GifExpertApp = ( { defaultCategories = [] }) => {

    // const [categories, setCategories] = useState(['cats', 'programing']);
    const [categories, setCategories] = useState(defaultCategories);

    return (<> 
        <h1 id="title">Gifs search</h1>

        <AddCategory setCategories={setCategories}/>

        <ol>
            {
                categories.map((category, i) => 
                    <GiftGrid  key={category + i } category={category}/>
                )
            }
        </ol>
     </>);
};

// export default GifExpertApp;
