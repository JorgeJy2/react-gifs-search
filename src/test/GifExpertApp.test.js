import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('GifExpertApp', ()=> {
    test('wraper', () => {
        const wraper = shallow(<GifExpertApp />);
        expect(wraper).toMatchSnapshot();
    });

    test('should view list categories ', ()=> {
        const categories = ['cats', 'programing'];
        const wraper = shallow(<GifExpertApp defaultCategories= {categories} />);
        expect(wraper).toMatchSnapshot();
        expect(wraper.find('GiftGrid').length).toBe(categories.length);
    });
});