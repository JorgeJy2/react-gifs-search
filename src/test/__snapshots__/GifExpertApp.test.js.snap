// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp should view list categories  1`] = `
<Fragment>
  <h1
    id="title"
  >
    Gifs search
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="cats"
      key="cats0"
    />
    <GiftGrid
      category="programing"
      key="programing1"
    />
  </ol>
</Fragment>
`;

exports[`GifExpertApp wraper 1`] = `
<Fragment>
  <h1
    id="title"
  >
    Gifs search
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
